Change the submenu flyout to match the height of the main menu + three more Justify content options for flexbox layouts

RonnieMendix opened the pull request:
Submenu opens on same height as main menu to make the submenu feel more as part of the menu instead of a seperate fly-out thing

## Checklist
- Contains unit tests ❌
- Contains breaking changes ❌
- Contains Atlas changes ✅ 
- Compatible with: MX 9️⃣

#### Web specific
- Contains e2e tests  ❌
- Is accessible ✅ ❌
- Compatible with Studio ✅
- Cross-browser compatible ✅ 

#### Native specific
- Works in Android ✅ 
- Works in iOS ✅ 
- Works in Tablet ✅ 

#### Feature specific
- Comply with designs ✅ 
- Comply with PM's requirements ✅ 

**_Please remove unnecessary emojis and sections and this comment before proceeding_**

## This PR contains
- [ ] Bug fix
- [x] Feature
- [ ] Refactor
- [ ] Documentation
- [ ] Other (describe)

## What is the purpose of this PR?
_Submenu opens on the same height as the main menu to make the submenu feel more a part of the menu instead of a separate fly-out thing_

## Relevant changes
_changed the top height of the submenu flyout to match the height of the top menu_

## What should be covered while testing?
_If the submenu indeed starts under the top menu on wide enough screen (high resolutions) and that the submenu still acts as it did on all other resolutions (smaller screens)_

## Extra comments (optional)
_Please add extra comments or delete the section if not required_

RonnieMendix committed:
Update _layout-atlas-responsive.scss

Submenu opens on same height as main menu to make the submenu feel more as part of the menu instead of a seperate fly-out thing

RonnieMendix committed:
Update _layout-atlas-responsive.scss

Submenu opens on same height as main menu to make the submenu feel more as part of the menu instead of a seperate fly-out thing

RonnieMendix committed:
Update for justifying content

Added 3 justify content options using flexbox values that where still missing.
These justify options are part of the basic support of flexbox

RonnieMendix committed:
Update for justifying content

Added 3 justify content options using flexbox values that where still missing.
These justify options are part of the basic support of flexbox

diego-antonelli commented on packages/theming/atlas/src/themesource/atlas_core/web/layouts/_layout-atlas-responsive.scss:
```diff
@@ -23,7 +23,7 @@
                         position: absolute;
                         z-index: 100;
-                        top: 0;
+                        top: $topbar-minimalheight;
```
@RonnieMendix this seems to be very break as it will start with a margin top as the same height as the topbar but related to the menu, which has its own position: relative. Can you explain me why this change is necessary?

RonnieMendix replied:
The menu does not have its own position relative… So it will look at the relativeness of the whole screen, and then you can add the top distance the same as the topbar.

But ow you mention it, what we also can do, is adding position: relative; to the mx-scrollcontainer-middle, because that is where the menu is located. Then top: 0; for the submenu will also work and show the submenu under the topbar. (which probably is more safe if you don't want a topbar and doesn't set the variable [topbar height] to 0)

RonnieMendix replied:
@diego-antonelli  Do you want me to commit a version with position relative instead of the topbar height as top spacing?

diego-antonelli reviewed (changes requested):
Thx for your contribution again. I added a comment, please reply in here.

Also, `Update _layout-atlas-responsive.scss` is not a good title for a PR, be more specific.

Commit:
fix: execution of modules scripts

Signed-off-by: Diego Antonelli <diego.antonelli@mendix.com>

Commit:
fix: execution of modules scripts

Signed-off-by: Diego Antonelli <diego.antonelli@mendix.com>

RonnieMendix committed:
Submenu height the same as mainmenu height v2

Submenu opens on same height as main menu to make the submenu feel more as part of the menu instead of a seperate fly-out thing without top spacing but with position relative on parent layout container

RonnieMendix committed:
Submenu height the same as mainmenu height v2

Submenu opens on same height as main menu to make the submenu feel more as part of the menu instead of a seperate fly-out thing without top spacing but with position relative on parent layout container

RonnieMendix committed:
Submenu height the same as mainmenu height v2

Submenu opens on same height as main menu to make the submenu feel more as part of the menu instead of a seperate fly-out thing without top spacing but with position relative on parent layout container

RonnieMendix committed:
Revert "Submenu height the same as mainmenu height v2"

This reverts commit 1116a487a08059e3f0993481c25be91e148e34d4.

RonnieMendix committed:
Revert "Submenu height the same as mainmenu height v2"

This reverts commit 1116a487a08059e3f0993481c25be91e148e34d4.

RonnieMendix committed:
Revert "Submenu height the same as mainmenu height v2"

This reverts commit 1116a487a08059e3f0993481c25be91e148e34d4.

Commit:
fix: remove execution of shared/charts from build and release step

Signed-off-by: Diego Antonelli <diego.antonelli@mendix.com>

Commit:
fix: remove execution of shared/charts from build and release step

Signed-off-by: Diego Antonelli <diego.antonelli@mendix.com>

leonardomendix committed:
test: fix video player and gallery screenshot baseline

RonnieMendix committed:
Card customization improvement

This will give the customer more options to manage their card styling
a card has it's own border-radius styling and the default card background color can be set in the custom-variables

RonnieMendix committed:
Card customization improvement

This will give the customer more options to manage their card styling
a card has it's own border-radius styling and the default card background color can be set in the custom-variables

fix: execution of modules scripts

Signed-off-by: Diego Antonelli <diego.antonelli@mendix.com>

fix: remove execution of shared/charts from build and release step

Signed-off-by: Diego Antonelli <diego.antonelli@mendix.com>

test: fix video player and gallery screenshot baseline

Gradient now uses variables to color itself

The gradient had fixed colors set as gradient values.  This gives customisation more work, cause if your main theme color isn't blue, then you need to change the 5 values of the gradient to your new color scheme, while you could just use the already built-in color scheme's based upon the primary color.

Update _layout-atlas-responsive.scss

Submenu opens on same height as main menu to make the submenu feel more as part of the menu instead of a seperate fly-out thing

Update for justifying content

Added 3 justify content options using flexbox values that where still missing.
These justify options are part of the basic support of flexbox

Submenu height the same as mainmenu height v2

Submenu opens on same height as main menu to make the submenu feel more as part of the menu instead of a seperate fly-out thing without top spacing but with position relative on parent layout container

Revert "Submenu height the same as mainmenu height v2"

This reverts commit 1116a487a08059e3f0993481c25be91e148e34d4.

Submenu height the same as mainmenu height v3

Submenu opens on same height as main menu to make the submenu feel more as part of the menu instead of a seperate fly-out thing
v3 = without top spacing but with position relative on parent layout container

Submenu height the same as mainmenu height v4

Submenu opens on same height as main menu to make the submenu feel more as part of the menu instead of a seperate fly-out thing without top spacing but with position relative on parent layout container

RonnieMendix committed:
Gradient now uses variables to color itself

The gradient had fixed colors set as gradient values.  This gives customisation more work, cause if your main theme color isn't blue, then you need to change the 5 values of the gradient to your new color scheme, while you could just use the already built-in color scheme's based upon the primary color.

RonnieMendix committed:
Submenu height the same as mainmenu height v3

Submenu opens on same height as main menu to make the submenu feel more as part of the menu instead of a seperate fly-out thing
v3 = without top spacing but with position relative on parent layout container

RonnieMendix committed:
Submenu height the same as mainmenu height v3

Submenu opens on same height as main menu to make the submenu feel more as part of the menu instead of a seperate fly-out thing
v3 = without top spacing but with position relative on parent layout container

RonnieMendix committed:
Submenu height the same as mainmenu height v3

Submenu opens on same height as main menu to make the submenu feel more as part of the menu instead of a seperate fly-out thing
v3 = without top spacing but with position relative on parent layout container

diego-antonelli commented on packages/theming/atlas/src/themesource/atlas_core/web/layouts/_layout-atlas-responsive.scss:
```diff
@@ -10,60 +10,63 @@
                             position: absolute;
                             z-index: 100;
-                            top: 0;
+                            top: $topbar-minimalheight;
```
This should be changed back to 0 as you added position relative to the parent

RonnieMendix committed:
Submenu height the same as mainmenu height v4

Submenu opens on same height as main menu to make the submenu feel more as part of the menu instead of a seperate fly-out thing without top spacing but with position relative on parent layout container

RonnieMendix committed:
Submenu height the same as mainmenu height v4

Submenu opens on same height as main menu to make the submenu feel more as part of the menu instead of a seperate fly-out thing without top spacing but with position relative on parent layout container

RonnieMendix committed:
Submenu height the same as mainmenu height v4

Submenu opens on same height as main menu to make the submenu feel more as part of the menu instead of a seperate fly-out thing without top spacing but with position relative on parent layout container

RonnieMendix committed:
added release notes to this release

### Changed
- Submenu on the navsidebar desktop now opens on the same height as the main menu

RonnieMendix committed:
added release notes to this release

### Changed
- Submenu on the navsidebar desktop now opens on the same height as the main menu

RonnieMendix committed:
Merge branch 'master' of https://github.com/mendix/widgets-resources

# Conflicts:
#	packages/modules/atlas-core/CHANGELOG.md

RonnieMendix committed:
Merge branch 'master' of https://github.com/mendix/widgets-resources

RonnieMendix committed:
Merge branch 'master' of https://github.com/RonnieMendix/widgets-resources

RonnieMendix committed:
changelog item added for the fixed i did

Becaus eof the update of my repository i lost my addition in the changelog. So now i added it again

RonnieMendix committed:
changelog item added for the fixed i did

Becaus eof the update of my repository i lost my addition in the changelog. So now i added it again

diego-antonelli commented:
@RonnieMendix can you squash commits? 22 commits for the changes is a bit too much

RonnieMendix committed:
Merge branch 'master' of https://github.com/RonnieMendix/widgets-resources

RonnieMendix commented:
I think i Squashed it now 🤔

tamelbrun commented on packages/modules/atlas-core/CHANGELOG.md:
```diff
@@ -4,6 +4,11 @@ All notable changes to this widget will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/), and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
 ## [Unreleased]
+### Changed
+- We fixed that the navsidebar sub-menu starts at the same height as the sidemenu itself
+
+### Changed
+- Submenu on the navsidebar desktop now opens on the same height as the main menu
```
The **Changed** section is now duplicated, please rewrite the commit which introduces this.

tamelbrun reviewed (changes requested):
I'm seeing multiple merge commits _into_ your branch; we don't do that, we want to keep our history clean. It also seems that there are now even more commits than before your attempt to squash, some of which appear identical. It might be better to replay your changes from scratch in a single commit.
